### ISSUE TITLE
Controller Got Deleted

### ISSUE THREAD

**Issue by creative-hustlers** [jonataslaw/getx#1586]
**Describe the bug**
I have 3 Screens Screen A, Screen B, Screen C
I want to navigate like this A->B->C then from C I want to delete all previous navigation route and need to go to A and reinitialize Controller of Screen A

but when we use  Get.offAllNamed(RouterName.A) it will delete the controller of screen A and not create new one, please note that I am using following for adding dependency

```
class ScreenABinding extends Bindings {
  @override
  void dependencies() {
    Get.lazyPut(() => ScreenAController());
  }
}
```

using fenix also not work

**Reproduction code
FROM Page A to B

Get.toNamed(RouterName.ScreenB);

FROM Page B to C

Get.toNamed(RouterName.ScreenC);

FROM Page C to A

Get.offAllNamed(RouterName.ScreenA);

**Expected behavior**
Get.offAllNamed remove old routed and create new page with new controller

**Flutter Version:**
Flutter 2.2.1 

**Getx Version:**
4.1.2

**Describe on which device you found the bug:**
Android.

**Comment by creative-hustlers** [comment]
#1461 same issue

**Comment by creative-hustlers** [comment]
#1349 same issue

**Comment by creative-hustlers** [comment]
#904 same issue

**Comment by creative-hustlers** [comment]
It seems this issue is from long time please provide proper solution for this issue, Thanks!

**Comment by creative-hustlers** [comment]
Same issue `https://stackoverflow.com/questions/66138542/flutter-getx-get-back-or-navigator-pop-deletes-controller-from-memory-and-ca` tried using **fenix** but also dosen't work

**Comment by jefrisingh** [comment]
instead of stateless widget try to use GetView<YourControllerClass> or GetWidget<YourControllerClass> also use Get.create instead of Get.put

**Comment by jefrisingh** [comment]
Check the getx doc about GetView and GetWidget for more info

**Comment by Drolegc** [comment]
I have the same issue
```
[GETX] GOING TO ROUTE /home
[GETX] REMOVING ROUTE /pregunta/4
[GETX] REMOVING ROUTE /categoria/1
[GETX] REMOVING ROUTE /home
[GETX] "ParticipanteHomeController" onDelete() called
[GETX] "ParticipanteHomeController" deleted from memory
```

But when I do the same flow, like, going to page 2 and return with offAllNamed, it works properly

```
[GETX] GOING TO ROUTE /home
[GETX] REMOVING ROUTE /pregunta/4
[GETX] REMOVING ROUTE /categoria/1
[GETX] REMOVING ROUTE /home
[GETX] Instance "ParticipanteHomeController" has been created
[GETX] Instance "ParticipanteHomeController" has been initialized
```

With the fenix flag, the ParticipanteHomeController does not get eliminated. But in this case, I don't know how good could it be, because when "it works properly", says that the controller has been created (again?)

**Comment by creative-hustlers** [comment]
> Check the getx doc about GetView and GetWidget for more info

I am currently using this only, still no luck and **GetView** also extends Stateless Widget so basically they are same.

**Comment by creative-hustlers** [comment]
@jefrisingh do you have any working solution for this problem, if yes kindly share here

**Comment by Drolegc** [comment]
If I change GET.offAllNamed for this
`                      Get.until((route) => (route.settings.name == Routes.PARTICIPANTE_HOME.toRoute()));
`
It seems to work, and Im using **fenix:true**

**Comment by quocviet1996** [comment]
Same issue, but i can't use fenix:true so i need something like we can control re-create controller  or fix this issue so the controller can re-create automatically.

**Comment by mrgithub** [comment]
How do you manually re-intialize the controller so it behaves as if it's freshly created and drops all the old values?

**Comment by creative-hustlers** [comment]
@mrgithub currently there is no way to do it

**Comment by creative-hustlers** [comment]
> How do you manually re-intialize the controller so it behaves as if it's freshly created and drops all the old values?

currently there is no way to do it

**Comment by Drolegc** [comment]
Apparently this was solved on the new version of GetX (get: ^4.3.0)

**Comment by jonataslaw** [comment]
I can't play in the latest version, so closing it
